chia/rpc/rpc_server.py

```diff
@@ -259,30 +258,30 @@ async def connection(self, ws):
 
                 break
 
-        await ws.close()
-
     async def connect_to_daemon(self, self_hostname: str, daemon_port: uint16):
-        while True:
+        while not self.shut_down:
             try:
-                if self.shut_down:
-                    break
-                async with aiohttp.ClientSession() as session:
-                    async with session.ws_connect(
-                        f"wss://{self_hostname}:{daemon_port}",
-                        autoclose=True,
-                        autoping=True,
-                        heartbeat=60,
-                        ssl_context=self.ssl_context,
-                        max_msg_size=100 * 1024 * 1024,
-                    ) as ws:
-                        self.websocket = ws
-                        await self.connection(ws)
-                    self.websocket = None
+                self.client_session = aiohttp.ClientSession()
+                self.websocket = await self.client_session.ws_connect(
+                    f"wss://{self_hostname}:{daemon_port}",
```

Comment:
Wonder why you changed this code as I think the "with" pattern is preferred in general as cleanup is handled for you

Reply:
We were having some issues with `with`, I believe it might be related to saving the object to the class in with, and reusing it later.

Reply:
This also potentially fixes the annoying error logging when shutting down the wallet.

Reply:
Yeah it seems to fix the not closed error, and I think it looks better for this use case